libminifi/test/flow-tests/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+file(GLOB FLOW_TESTS  "*.cpp")
+SET(FLOW_TEST_COUNT 0)
+FOREACH(testfile ${FLOW_TESTS})
+    get_filename_component(testfilename "${testfile}" NAME_WE)
+    add_executable("${testfilename}" "${testfile}")
+    createTests("${testfilename}")
+    target_link_libraries(${testfilename} ${CATCH_MAIN_LIB})
+    target_wholearchive_library(${testfilename} minifi-standard-processors)
+    MATH(EXPR FLOW_TEST_COUNT "${FLOW_TEST_COUNT}+1")
+    add_test(NAME "${testfilename}" COMMAND "${testfilename}" WORKING_DIRECTORY ${TEST_DIR})
+ENDFOREACH()
+message("-- Finished building ${FLOW_TEST_COUNT} flow related test file(s)...")
```

Comment:
The file should end with a newline character

Reply:
done

Reply:
(it seems like the Enter puts too much stress on my pinky)

Reply:
(some more bikeshedding questions about the POSIX "line")

<img width="170" alt="Screenshot 2020-07-06 at 8 32 31" src="https://user-images.githubusercontent.com/64783590/86563354-85c2ea80-bf64-11ea-8f3a-a5c2ac5f66f8.png">

what does POSIX say about where the cursor would be? it cannot be on "line 2" as there is no "line 2" 

or take the error message:

<img width="599" alt="Screenshot 2020-07-06 at 8 33 49" src="https://user-images.githubusercontent.com/64783590/86563648-097cd700-bf65-11ea-87cb-d8d9ebf12ddd.png">

what is the `2` in `test.cpp:**2**:8`?

Reply:
The tools were written to behave decently with non-text files and consider the line number to be the encountered `\n`s + 1. Historical posix text processing utilities may ignore the contents after the last `\n`, but I haven't seen modern tools do so. [1]

Please note that I'm by no means a POSIX lawyer, and the part about handling is mostly speculation.

[1] https://unix.stackexchange.com/a/18789/366131

Reply:
`text-file` is a file format that has the magic number `\n` at the end of the file 🤯

Reply:
I'm not sure about the above implication (`\n` before EOF -> is text file), but its contrary (no `\n` before EOF -> is not a text file) seems to be true to me.

Reply:
this is a response to a deleted comment about how `text file` is a file format with the magic number `\n` at the end of the file (I don't know how to resurrect the comment)